bumpversion/cli.py

```diff
@@ -66,6 +67,42 @@
 ]
 
 
+def main(original_args=None):
+    args, known_args, root_parser, positionals = _parse_arguments_phase_1(original_args)
+    _setup_logging(known_args.list, known_args.verbose)
+    vcs_info = _determine_vcs_usability()
+    defaults = _determine_current_version(vcs_info)
+    explicit_config = None
+    if hasattr(known_args, "config_file"):
+        explicit_config = known_args.config_file
+    config_file = _determine_config_file(explicit_config)
+    config, config_file_exists, config_newlines, part_configs, files = _load_configuration(
+        config_file, explicit_config, defaults,
+    )
+    known_args, parser2, remaining_argv = _parse_arguments_phase_2(
+        args, known_args, defaults, root_parser
+    )
+    vc = _setup_versionconfig(known_args, part_configs)
+    current_version = _parse_current_version(known_args.current_version, vc)
+    context = _assemble_context(vcs_info)
+    new_version = _assemble_new_version(
+        context, current_version, defaults, known_args.current_version, positionals, vc
+    )
+    args, file_names = _parse_arguments_phase_3(remaining_argv, positionals, defaults, parser2)
+    new_version = _parse_new_version(args, new_version, vc)
+    _determine_files(file_names, files, positionals, vc)
```

Comment:
This method doesn't really match its description. How about doing it inline:
```suggestion
    files += [ConfiguredFile(file_name, vc) for file_name in (file_names or positionals[1:])]
```